android/java/org/mlperf/inference/MLPerfTasks.java

```diff
@@ -115,25 +119,39 @@ public static boolean isZipFile(String path) {
     return path.endsWith(ZIP);
   }
 
+  public static String getCacheDirName() {
+    return LOCAL_CACHE_FOLDER;
+  }
+
   public static String getLocalPath(String path) {
 
     // Use the localDir's cache directory path only if its a github link or assets link.
     // For local files, use it as is.
     String filename = new File(path).getName();
-    if (path.startsWith("http://") || path.startsWith("https://") || isZipFile(filename)) {
+    boolean isURL = path.startsWith("http://") || path.startsWith("https://");
+    if (isURL || isZipFile(filename)) {
       if (isZipFile(filename)) {
         filename = filename.substring(0, filename.length() - ZIP.length());
       }
-      String localDir =
-          MLCtx.getInstance().getContext().getExternalFilesDir("cache").getAbsolutePath();
-      return localDir + "/cache/" + filename;
+
+      if (isURL) {
+        String longFileName;
+        try{
+          longFileName = new URL(path).getPath();
+        }
+        catch(Exception e) {
+          longFileName = filename;
```

Comment:
What failure condition of URL(path).getPath() will this fix?

Reply:
Does File(path).getName() of a URL produce the expected result?

Reply:
First "if" drops ".zip" extension if presented, second "if" keep "path" from URL.
The resulting name of the file will be something /{cache directory path}/{file name} for locally stored file
and /{cache directory path}/{path from url}/{file name} for URL. In both cases ".zip" extension will be dropped if presented. So if downloaded files have the same name, but different URLs - they will be stored in different folders in the cache

Reply:
The code seems hard to understand for a quick scan. BTW, the path of some folders still have a `.zip` suffix like this one: `/sdcard/mlperf_results/mlperf/cache/mlcommons/mobile_models/raw/main/v0_7/datasets/imagenet.zip`